doc/source/dev/development_environment.rst

```diff
@@ -238,6 +238,23 @@ Most python builds do not include debug symbols and are built with compiler
 optimizations enabled. To get the best debugging experience using a debug build
 of Python is encouraged, see :ref:`advanced_debugging`.
 
+In terms of debugging, NumPy also needs to be built in a debug mode. You need to use
+``debug`` build type and disable optimizations to make sure ``-O0`` flag is used
+during object building. To generate source-level debug information within the build process run::
+
+    $ CFLAGS="-O0 -g" CXXFLAGS="-O0 -g" spin build --clean -- -Dbuildtype=debug -Ddisable-optimization=true
+
+.. note::
+
+    In case you are using conda environment be aware that conda sets ``CFLAGS``
+    and ``CXXFLAGS`` automatically, and they will include the ``-O2`` flag by default.
+    To take control of these variables in conda, you can create ``env_vars.sh``
```

Comment:
Maybe something like: "You can safely use `unset CFLAGS && unset CXXFLAGS` to avoid them. Alternatively, to take control of ... more permanently, ..."

Reply:
Sure! I updated it.